internal/atomiccounter/atomiccounter.go

```diff
@@ -0,0 +1,28 @@
+package atomiccounter
```

Comment:
Would be best to add a unit test in similar to this test case that I did for my semaphore benchmarking?
```
for _, c := range cases {
		t.Run(c.testName, func(t *testing.T) {
			sem := NewSemaphore(c.limit)

			channel := make(chan struct{})
			wg := sync.WaitGroup{}
			for i := 0; i < c.numberOfFiles; i++ {
				wg.Add(1)
				go func() {
					<-channel
					ok := sem.Acquire(time.Second)
					assert.True(t, ok)
					sem.Release()
					wg.Done()
				}()
			}

			close(channel) // start
			wg.Wait()

			checkLimitAndCount(t, sem, c.limit, 0)
		})
	}
```

Reply:
Added 2 unit test cases.